src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -102,6 +102,36 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                 {
                     // TODO: We need to handle format parameter.
                 }
+                else if (query.Item1 == KnownQueryParameterNames.Type)
+                {
+                    var types = query.Item2.SplitByOrSeparator();
+                    var badTypes = types.Where(type => !ModelInfoProvider.IsKnownResource(type)).ToHashSet();
+
+                    if (badTypes.Count != 0)
+                    {
+                        _contextAccessor.FhirRequestContext.BundleIssues.Add(
+                            new OperationOutcomeIssue(
+                                OperationOutcomeConstants.IssueSeverity.Information,
+                                OperationOutcomeConstants.IssueType.Informational,
+                                string.Format(Core.Resources.InvalidTypeParameter, string.Join(",", badTypes))));
+                        if (badTypes.Count != types.Count)
+                        {
+                            searchParams.Add(KnownQueryParameterNames.Type, string.Join(',', types.Except(badTypes)));
+                            foreach (var badType in badTypes)
+                            {
+                                unsupportedSearchParameters.Add(new Tuple<string, string>(KnownQueryParameterNames.Type, badType));
```

Comment:
So just confirming that if I passed in `?_type=Observation,foo1,foo2` then the resulting Bundle self link would have `?_type=Observation`?

Reply:
It is based on routing + the behavior of dropping unsupported options, have a look at UrlResolver to see if its consistent 
https://github.com/microsoft/fhir-server/blob/f96a3bc70da1422fb8de8e0129cba7e7e3318f02/src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs#L144-L236

Reply:
I guess I broke contract of what unsupported parameters should mean.
One in `UrlResolver` looks for exact matching, while I put pieces instead of whole.

Example: `_type=Observation,foo,foo1` would have two unsupported parameters `_type=foo` and `_type==foo1`.
But in case if I pass `_type=foo` it would be removed from link.

What is preferred behavior here?


---
In reply to: [511157028](https://github.com/microsoft/fhir-server/pull/1385#discussion_r511157028) [](ancestors = 511157028)

Reply:
I think the ideal case is that the unsupported options are removed from the link

Reply:
I've added logic to remove unsupported types and add tests for that.

---
In reply to: [512828294](https://github.com/microsoft/fhir-server/pull/1385#discussion_r512828294) [](ancestors = 512828294)

Reply:
Change it to contain only valid types in self link

---
In reply to: [511147213](https://github.com/microsoft/fhir-server/pull/1385#discussion_r511147213) [](ancestors = 511147213,511142603)